### PR TITLE
Reorganize modules

### PR DESCRIPTION
**Pull request by SantiagoTorres**
It'll be very hard to work on the code by having all of us touch a single file. The functionality should be split into separate modules. I did some general reorganization based on some intuition. Please see commit logs/messages to make sense of it.

### CODE REVIEW

File: privateSBOMExchange/src/scripts/Redaction.py [hunk 1]
```diff
@@ -241,9 +46,9 @@ def main():
     #SBOM_file_name=str(args.sbom)
  
     #  path of submodules data 
-    in_the_lab_spdx="../sbom_data/bom-shelter/in-the-lab/spdx-popular-containers/data"
-    in_the_wild_cyclonedx="../sbom_data/bom-shelter/in-the-wild/cyclonedx"
-    in_the_wild_spdx = '../sbom_data/bom-shelter/in-the-wild/spdx' 
+    in_the_lab_spdx="../../sbom_data/bom-shelter/in-the-lab/spdx-popular-containers/data"
+    in_the_wild_cyclonedx="../../sbom_data/bom-shelter/in-the-wild/cyclonedx"
+    in_the_wild_spdx = '../../sbom_data/bom-shelter/in-the-wild/spdx'
```
Comment (marcelamelara) [inline privateSBOMExchange/src/scripts/Redaction.py:51]: These paths should eventually be moved out into config files.

File: privateSBOMExchange/src/scripts/Redaction.py [hunk 2]
```diff
@@ -241,9 +46,9 @@ def main():
...
     #  path of submodules data 
-    in_the_lab_spdx="../sbom_data/bom-shelter/in-the-lab/spdx-popular-containers/data"
-    in_the_wild_cyclonedx="../sbom_data/bom-shelter/in-the-wild/cyclonedx"
-    in_the_wild_spdx = '../sbom_data/bom-shelter/in-the-wild/spdx' 
+    in_the_lab_spdx="../../sbom_data/bom-shelter/in-the-lab/spdx-popular-containers/data"
+    in_the_wild_cyclonedx="../../sbom_data/bom-shelter/in-the-wild/cyclonedx"
+    in_the_wild_spdx = '../../sbom_data/bom-shelter/in-the-wild/spdx' 
 
 
     SBOM_file_name="julia.spdx.json"
```
Comment (marcelamelara) [inline privateSBOMExchange/src/scripts/Redaction.py:54]: Same with this and any hardcoded filenames, these should all be passed in via a config file or CLI args.

File: privateSBOMExchange/src/petra/lib/sbom/spdx.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            out[name[:-1]] = x
+
+    flatten(y)
+    return out
+
+
+def flatten_SPDX(file_name):
+    with open (file_name, 'r') as sbom_file:
+        sbom_file_encoding=sbom_file.encoding
+        sbom_data = json.load(sbom_file)
```
Comment (marcelamelara) [inline privateSBOMExchange/src/petra/lib/sbom/spdx.py:24]: Shouldn't we be catching an exception here?

File: privateSBOMExchange/src/petra/lib/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                SBOMField[item]=str(SBOMField[item])
+            root1 = tree.update(item.encode(sbom_file_encoding), SBOMField[item].encode(sbom_file_encoding))
+            assert 32 == len(root1)
+            assert root1 != PLACEHOLDER
+        
+    return tree, tree_name
+
+
+
+def try_tree():
```
Comment (marcelamelara) [inline privateSBOMExchange/src/petra/lib/models/__init__.py:30]: Is this test code? This should be moved to a separate test suite.

File: privateSBOMExchange/src/petra/lib/models/__init__.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    for field_name,value in flatten_SBOM_data.items():
+        count+=1
+        SBOMField={field_name:value}
+        if field_name =="name":
+            tree_name=value
+        assert DEFAULTVALUE == tree.get(b"")
+        for item in SBOMField:
+            if isinstance(SBOMField[item], bool):
+                SBOMField[item]=str(SBOMField[item])
+            root1 = tree.update(item.encode(sbom_file_encoding), SBOMField[item].encode(sbom_file_encoding))
```
Comment (marcelamelara) [inline privateSBOMExchange/src/petra/lib/models/__init__.py:22]: What is this line doing? Why are we encoding the contents of the tree based on the SBOM file encoding? I think we should be operating on raw bytes.

File: privateSBOMExchange/src/petra/lib/sbom/spdx.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import json 
+
+def flatten_data(y):
```
Comment (marcelamelara) [inline privateSBOMExchange/src/petra/lib/sbom/spdx.py:3]: How many SBOMs has this been tested on?

File: privateSBOMExchange/src/scripts/Redaction.py [hunk 7]
```diff
@@ -3,231 +3,36 @@
...
-            if isinstance(value, bool):
-                value=str(value)
-            SBOM_field_file.write(value)                     
-        args = make_args(f"3 {pub_key} {key} {policy}")
-        print(*args)
-        cpabe_encrypt_functions.main(len(args), args)
-        cpabe_encrypt_functions.reset_globals()
-
-def  compare_stored_and_retrieved_tree(original,retrieved):
+def compare_stored_and_retrieved_tree(original,retrieved):
```
Comment (marcelamelara) [inline privateSBOMExchange/src/scripts/Redaction.py:26]: This seems like another test function that should be moved out into the test suite.

File: privateSBOMExchange/src/scripts/Redaction.py [hunk 8]
```diff
@@ -260,7 +65,7 @@ def main():
 
     
 
-    #using tree stucture, store sbom as tree in sqlalchamey db
+    #using tree stucture, store sbom as tree in sqlalchemy db
     t1=try_tree()
```
Comment (marcelamelara) [inline privateSBOMExchange/src/scripts/Redaction.py:69]: I understand that this is a work in progress, but we really shouldn't be mixing testing with core system APIs. The Redaction CLI should do only one thing: take an SBOM and output a redacted version.

**Review by marcelamelara — COMMENTED** [review]
Thanks for these changes @SantiagoTorres !! 

I left a bunch of comments that we can hopefully address in future PRs. @abuishgair Please open issues to track my requested changes. I'll approve this PR once those issues are opened.

**Review by abuishgair — COMMENTED** [review]
Thank you @SantiagoTorres!
I did review the structure and added seperate issues for @marcelamelara comments.

**Comment by marcelamelara** [comment]
> I did review the structure and added seperate issues for @marcelamelara comments.

Thanks @abuishgair !!

**Review by marcelamelara — APPROVED** [review]

**Pull request merged**